geopandas/geodataframe.py

```diff
@@ -888,8 +888,11 @@ def iterfeatures(self, na="null", show_bbox=False, drop_id=False):
                 geom = geometries[i]
 
                 if na == "drop":
+                    na_mask = pd.isna(row)
```

Comment:
You could also construct this NA mask for all rows up front using `properties[properties_cols].isna()` then indexing into this dataframe with `i` - as was suggested in the original issue. I would imagine that is slightly faster, but I could be wrong.

Reply:
Really appreciate the review @m-richards, I assume you mean something like `pd.isna(properties)` ([see new commit](https://github.com/geopandas/geopandas/pull/2811/commits/16b6b8bcf29deeda4dbf63e3737840cecc1532fa))? The property columns are already assigned to `properties` unless I misunderstood.

Reply:
Sorry for the confusion, `self[properties_cols].isna()` is what I meant to type, but what you've implemented in terms of `properties` directly is probably better.